test/test_checkpoint.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License
+# is located at
+#
+#     http://aws.amazon.com/apache2.0/
+# 
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from math import isclose
+import mxnet as mx
+import random
+import tempfile
+
+import sockeye.data_io
+
+def generate_random_sentence(vocab_size, max_len):
+    length = random.randint(1, max_len)
+    sentence = []
+    for _ in range(length):
+        sentence.append(random.randint(3, vocab_size + 2))
+    return sentence
+
+def create_ParallelBucketSentenceIter(source_sentences, target_sentences, max_len):
+    buckets = sockeye.data_io.define_parallel_buckets(max_len, 10)
+    batch_size = 50
+    eos = 0
+    pad = 1
+    unk = 2
+    bucket_iterator = sockeye.data_io.ParallelBucketSentenceIter(source_sentences,
+                                                         target_sentences,
+                                                         buckets,
+                                                         batch_size,
+                                                         eos, pad, unk)
+    return bucket_iterator
+
+def data_batches_equal(db1, db2):
+    # We just compare the data, should probably be enough
+    equal = True
+    for data1, data2 in zip(db1.data, db2.data):
+        equal = equal \
+                and (data1.shape == data2.shape) \
+                and isclose(mx.nd.prod(data1.reshape((-1,)) == data2.reshape((-1,))).asnumpy()[0], 1.0)
```

Comment:
Yes, but it converts again into "float booleans" (i.e. 0.0 for false, 1.0 for true). (Actually it should be safe to compare against 1.0)